File: DSA/DataStructures/Array/6. ValidAnagram/Solution1.js
// https://leetcode.com/problems/valid-anagram/description/

var isAnagram = function(s, t) {

    if(s.length!==t.length) return false

    let mp = {}
    
    for(let i=0; i<s.length; i++) {
        if(mp[s[i]]) {
            mp[s[i]]+=1
        } else {
            mp[s[i]]=1
        }
    }

    for(let i=0; i<t.length; i++) {
        if((mp[t[i]] == undefined) || (mp[t[i]] == 0) ) {
            return false
        } else {
            mp[t[i]]-=1
        }
    }

    return true
    
};

File: DSA/DataStructures/Array/8. MergeSort/Solution1.java
// https://leetcode.com/problems/merge-sorted-array/description/

class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {

        int l = m+n;
        l--;
        n--;
        m--;


        while(n>=0) {

            if(m>=0 && nums1[m]>nums2[n]) {
                nums1[l]=nums1[m];
                m--;
            } else {
                nums1[l]=nums2[n];
                n--;
            }
            l--;
        }
    }
}